octopoes/octopoes/api/bulk_router.py

```diff
@@ -49,3 +51,35 @@ def list_reports(
             reports[recipe_id] = report
 
     return reports
+
+
+@router.get("/object-clients")
+def list_object_clients(
+    clients: set[str] = Query(set()),
+    reference: Reference = Depends(extract_reference),
+    settings_: Settings = Depends(settings),
+    valid_time: datetime = Depends(extract_valid_time),
+) -> list[str]:
+    """
+    An efficient endpoint for checking if OOIs live in multiple organizations
+    """
+
+    # See list_reports() for some of the reasoning behind the below code
+    event_manager = EventManager("null", str(settings_.queue_uri), celery_app, QUEUE_NAME_OCTOPOES)
+    xtdb_http_client = get_xtdb_client(str(settings_.xtdb_uri), "")
+
+    clients_with_reference = []
+
+    for client in clients:
+        logger.info("client %s", client)
+        xtdb_http_client.client = client
+        ooi_repository = XTDBOOIRepository(event_manager, XTDBSession(xtdb_http_client))
```

Comment:
You can also move this `ooi_repository` variable outside the loop, this way you don't have to instantiate `XTDBOOIRepository` and `XTDBSession` on every iteration. Since the created repository already has an instance of the XTDB HTTP client; you only have to change the `client` property/ attribute in the loop like line 75